iis-common/src/main/java/eu/dnetlib/iis/common/java/io/SequenceFileTextValueReader.java

```diff
@@ -0,0 +1,168 @@
+package eu.dnetlib.iis.common.java.io;
+
+import java.io.IOException;
+import java.util.regex.Pattern;
+
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.LocatedFileStatus;
+import org.apache.hadoop.fs.RemoteIterator;
+import org.apache.hadoop.io.SequenceFile;
+import org.apache.hadoop.io.SequenceFile.Reader;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.io.Writable;
+import org.apache.hadoop.util.ReflectionUtils;
+import org.apache.log4j.Logger;
+
+/**
+ * Iterator that extracts sequence file's consecutive {@link Text} values. 
+ * 
+ * @author mhorst
+ */
+public class SequenceFileTextValueReader implements CloseableIterator<Text> {
+
+        private final Logger log = Logger.getLogger(this.getClass());
+
+        private SequenceFile.Reader sequenceReader;
+
+        private final RemoteIterator<LocatedFileStatus> fileIt;
+
+        private final FileSystem fs;
+
+        /** Ignore file starting with underscore. Such files are also ignored by
+         * default by map-reduce jobs.
+         */
+        private final static String whitelistRegexp = "^(?!_).*";
+
+        private Text toBeReturned;
+
+        /**
+         * Default constructor.
+         * 
+         * @param fs hadoop filesystem
+         * @param seqPath sequence file path
+         * @throws IOException
+         */
+        public SequenceFileTextValueReader(final FileSystemPath path) 
+                        throws IOException {
+                this.fs = path.getFileSystem();
+                if (fs.isDirectory(path.getPath())) {
+                        fileIt = fs.listFiles(path.getPath(), false);
+                        sequenceReader = getNextSequenceReader();
+                } else {
+                        fileIt = null;
+                        sequenceReader = new Reader(fs.getConf(),
+                                        SequenceFile.Reader.file(path.getPath()));
+                }
+        }
+
+        final Reader getNextSequenceReader() throws IOException {
+                while (fileIt != null && fileIt.hasNext()) {
+                        LocatedFileStatus currentFileStatus = fileIt.next();
+                        if (isValidFile(currentFileStatus)) {
+                                return new Reader(this.fs.getConf(),
+                                                SequenceFile.Reader.file(currentFileStatus.getPath()));
+                        }
+                }
+//              fallback
+                return null;
+        }
+
+        /**
+         * Checks whether file is valid candidate.
+         * @param fileStatus
+         * @return true when valid, false otherwise
+         */
+        private final boolean isValidFile(LocatedFileStatus fileStatus) {
+                if (fileStatus.isFile()) {
+                        if (whitelistRegexp!=null) {
```

Comment:
Yeah, it's dull. I have resurrected `SequenceFileTextValueReader` from old svn revision (it was dropped when we switched to avro), probably in first version of this class regexp was optional and after changing it to constant this condition was not removed.

Fixed.